shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -96,7 +96,11 @@
   public static native boolean nativeGetIsSoftwareRenderingEnabled();
 
   @UiThread
-  public static native String nativeGetObservatoryUri();
+  public static String getObservatoryUri() {
+    return observatoryUri;
+  }
+
+  private static String observatoryUri;
```

Comment:
Maybe I'm missing something, but `observatoryUri` doesn't seem to be set anywhere?

Also, with regard to style, would you mind placing static fields above static methods?

Reply:
This field is set by the engine using JNI.  Added a comment.

Reply:
Based on your work with isolate ID, should we communicate this via channel instead?

Reply:
The observatory URI is a process-wide global and isn't associated with a specific FlutterView/DartExecutor that can receive the channel message